.github/actions/setup-woocommerce-monorepo/action.yml

```diff
@@ -0,0 +1,56 @@
+name: Setup WooCommerce Monorepo
+description: Handles the installation, building, and caching of the projects within the monorepo.
+
+inputs:
+  filter:
+    description: The PNPM filter used to decide what projects to install and build.
+    default: "*"
+  build:
+    description: Indicates whether or not the action should build any projects.
+    default: "true"
+  php-version:
+    description: The version of PHP that the action should set up.
+    default: "7.2"
+
+runs:
+  using: composite
+  steps:
+    - name: Setup PNPM
+      uses: pnpm/action-setup@10693b3829bf86eb2572aef5f3571dcf5ca9287d
```

Comment:
[GitHub's documentation on security hardening](https://github.com/woocommerce/woocommerce/pull/34607) encourages maintainers to pin commit hashes for actions in order to avoid the possibility of malicious code being shipped in an action. While it does make maintenance a bit more difficult, this is likely something we should be doing in _all_ of our actions.

Reply:
I guess doing this will also encourage us to make actions more modular too so its win/win :D